docs/src/content/docs/reference/configuration.mdx

```diff
@@ -65,6 +65,33 @@ type LogoConfig = { alt?: string; replacesTitle?: boolean } & (
 
 Configure the table of contents shown on the right of each page. By default, `<h2>` and `<h3>` headings will be included in this table of contents.
 
+### `expressiveCode`
+
+**type:** `StarlightExpressiveCodeOptions | boolean`
+
+Starlight uses [Expressive Code](https://github.com/expressive-code/expressive-code/tree/main/packages/astro-expressive-code) to render code blocks on your site. The integration is enabled by default and can be disabled by passing `false` to this option.
+
+You can customize your code blocks by passing an object with any of the following properties:
+
+- `themes` — an array of themes that should be used to render code blocks. The following item types are supported in this array:
+  - any theme name bundled with Shiki (e.g. `dracula`)
+  - any theme object compatible with VS Code or Shiki (e.g. imported from an NPM theme package)
+  - any ExpressiveCodeTheme instance (e.g. using `ExpressiveCodeTheme.fromJSONString(...)` to load a custom JSON/JSONC theme file yourself)
```

Comment:
Are you planning some kind of special Expressive Code support for more than 2 themes? I imagine the scenarios where this makes sense require some custom UI for picking the user-preferred colour theme?

Would there be an argument for making the Starlight option a little simpler and just have two `lightTheme` and `darkTheme` options? Or would the current array also automatically support e.g. light/dark/light high contrast/dark high contrast?

I’m also curious what actually happens if you provide more than two themes? Does only e.g. the first dark theme get used?

Reply:
Expressive Code supports any number of themes out of the box (not just "dark" and "light"), and I've already seen sites using this to provide additional theming options (non-Starlight if I remember correctly).

I've seen one site providing a three-state theme selector with options like "day", "sunset" and "night", and another adding a custom "high contrast" toggle that selects special versions of the dark and light themes when enabled (so effectively providing 4 themes in total, just like you mentioned in your comment).

I didn't want to limit this functionality to just two themes when EC is being used inside Starlight, so I just kept the same `themes` array option as provided by the upstream `astro-expressive-code` package.

If you provide more than two themes, the Starlight integration's `useStarlightDarkModeSwitch` option (which is enabled by default) will create CSS selectors that integrate with Starlight's default theme switcher by making the first dark theme available through `data-theme="dark"` and the first light theme through `data-theme="light"`, and all other themes through their individual names, e.g. `data-theme="my-custom-high-contrast-dark-theme"`.

Reply:
Thanks for the explanation!

I’m thinking a bit about that last case using `data-theme`. We don’t necessarily have to solve this in this PR — we have a chance to break the API in a future minor before Starlight v1 if needed — but because you can’t have `data-theme` set twice on an element, setting `<html data-theme="my-custom-high-contrast-dark-theme">` would presumably override the general Starlight theme (switching it to the dark theme, which is our default when `data-theme="light"` isn’t set). So basically Starlight theme and EC theme are tied together. Obviously in the general case dark/light switching with Starlight makes sense though.

I guess this the case that the `useStarlightDarkModeSwitch` option solves? You set that to `false` and decouple the switching?

Reply:
Yes, you set this option to false and decouple the switching, and then you can either just define your own CSS, or you use the `themeCssSelector` option to switch syntax highlighting themes based on something other than `data-theme`. You can use classes there, other data attributes etc. - allowing them to coexist with Starlight's dark/light mode switch without relying on the same attribute.

Reply:
Got it. OK! I think I’m happy to leave these two options as they are. I don’t think they’re super easy to understand or use for the advanced cases, but hopefully that’s a niche concern, and I don’t off the top of my head have any ideas on how to improve it 😁

Reply:
Ok, let's keep them like this for now then. If we or our users come up with better ideas in the future, I'm always happy to improve the options further. :)

Thank you for the thorough review and great discussions about the use cases!